Source/WebCore/Modules/permissions/PermissionStatus.cpp

```diff
@@ -31,48 +31,97 @@
 #include "DocumentInlines.h"
 #include "EventNames.h"
 #include "PermissionController.h"
+#include "PermissionState.h"
+#include "Permissions.h"
+#include "ScriptExecutionContext.h"
 #include "SecurityOrigin.h"
+#include "WorkerGlobalScope.h"
+#include "WorkerLoaderProxy.h"
+#include "WorkerThread.h"
+#include <memory>
+#include <wtf/HashMap.h>
 #include <wtf/IsoMallocInlines.h>
 #include <wtf/MainThread.h>
+#include <wtf/NeverDestroyed.h>
 
 namespace WebCore {
 
 WTF_MAKE_ISO_ALLOCATED_IMPL(PermissionStatus);
 
-Ref<PermissionStatus> PermissionStatus::create(ScriptExecutionContext& context, PermissionState state, const PermissionDescriptor& descriptor)
+static uint64_t nextMainThreadPermissionObserverIdentifier()
 {
-    auto status = adoptRef(*new PermissionStatus(context, state, descriptor));
+    static std::atomic<uint64_t> currentIdentifier;
+    return ++currentIdentifier;
+}
+
+static HashMap<uint64_t, std::unique_ptr<MainThreadPermissionObserver>>& allMainThreadPermissionObservers()
+{
+    static NeverDestroyed<HashMap<uint64_t, std::unique_ptr<MainThreadPermissionObserver>>> map;
+    return map;
+}
+
+static void addMainThreadPermissionObserver(uint64_t identifier, std::unique_ptr<MainThreadPermissionObserver>&& observer)
```

Comment:
Not sure this function is super helpful, why can't we just inline its body whenever it is used? It is one line.